### PR TITLE
Add check for multiple release in Namespace

### PR DESCRIPTION
**Pull request by knrt10**
User should know that before deleting a component that other helm
release  exists in a namespace.

closes: #1077
Signed-off-by: knrt10 <kautilya@kinvolk.io>

### CODE REVIEW

File: cli/cmd/component-delete.go [hunk 1]
```diff
@@ -20,6 +20,7 @@ import (
```
Comment (invidian) [inline cli/cmd/component-delete.go:20]: Commit message has double space.
```diff
- User should know that before deleting a component that other helm release  exists in a namespace.
+ User should know that before deleting a component that other helm release exists in a namespace.
```

File: cli/cmd/component-delete.go [hunk 2]
```diff
@@ -127,10 +128,32 @@ func componentNamesToObjects(componentNames []string) ([]components.Component, e
 	return c, nil
 }
 
+func checkMutlipleReleaseNamespace(ns string, kubeconfig []byte, deleteNamespace bool) error {
```
Comment (invidian) [inline cli/cmd/component-delete.go:132]: We should not add more code into `cli/cmd`. I'd make it part of `util.UninstallComponent`.

File: cli/cmd/component-delete.go [hunk 3]
```diff
@@ -127,10 +128,32 @@ func componentNamesToObjects(componentNames []string) ([]components.Component, e
...
+	if err != nil {
+		return fmt.Errorf("preparing helm client: %w", err)
+	}
+
+	releasesList, err := action.NewList(actionConfig).Run()
+	if err != nil {
+		return fmt.Errorf("listing helm releases: %w", err)
+	}
+
+	if len(releasesList) > 1 && deleteNamespace {
```
Comment (invidian) [inline cli/cmd/component-delete.go:143]: We should check for `>0` after we uninstall the release I think.

File: cli/cmd/component-delete.go [hunk 4]
```diff
@@ -127,10 +128,32 @@ func componentNamesToObjects(componentNames []string) ([]components.Component, e
 	return c, nil
 }
 
+func checkMutlipleReleaseNamespace(ns string, kubeconfig []byte, deleteNamespace bool) error {
+	actionConfig, err := util.HelmActionConfig(ns, kubeconfig)
+	if err != nil {
+		return fmt.Errorf("preparing helm client: %w", err)
```
Comment (invidian) [inline cli/cmd/component-delete.go:135]:
```suggestion
		return fmt.Errorf("preparing Helm client: %w", err)
```

File: cli/cmd/component-delete.go [hunk 5]
```diff
@@ -127,10 +128,32 @@ func componentNamesToObjects(componentNames []string) ([]components.Component, e
...
+	}
+
+	releasesList, err := action.NewList(actionConfig).Run()
+	if err != nil {
+		return fmt.Errorf("listing helm releases: %w", err)
```
Comment (invidian) [inline cli/cmd/component-delete.go:140]:
```suggestion
		return fmt.Errorf("listing Helm releases: %w", err)
```

File: cli/cmd/component-delete.go [hunk 6]
```diff
@@ -127,10 +128,32 @@ func componentNamesToObjects(componentNames []string) ([]components.Component, e
...
+	}
+
+	if len(releasesList) > 1 && deleteNamespace {
+		return fmt.Errorf("multiple release exists in: %q namespace", ns)
```
Comment (invidian) [inline cli/cmd/component-delete.go:144]:
```suggestion
		return fmt.Errorf("namespace %q may have other components installed, refusing to remove", ns)
```

**Review by invidian — CHANGES_REQUESTED** [review]

File: pkg/components/util/install.go [hunk 7]
```diff
@@ -205,7 +205,16 @@ func UninstallComponent(c components.Component, kubeconfig []byte, deleteNSBool
 		}
 	}
 
+	releasesList, err := action.NewList(cfg).Run()
```
Comment (invidian) [inline pkg/components/util/install.go:208]: Can we have e2e test installing 2 components into single namespace and then trying to remove them?

We could for example allow installing `httpbin` in separate namespace to make it easier, as this component should only create minimal amount of manifests (no not-namespaces resources etc).

**Review by invidian — CHANGES_REQUESTED** [review]
LGTM, but I would add tests for it.

File: pkg/components/util/install.go (hunk 7 above)
Comment (knrt10) [reply pkg/components/util/install.go:208]: We can test for, headlamp as it is already installed in `lokomotive-system` namespace.

Comment (invidian) [reply pkg/components/util/install.go:208]: Hm, but if we uninstall headlamp, then headlamp tests may fail. I guess we can run them as disruptive then.

**Review by invidian — APPROVED** [review]
LGTM

**Review by invidian — APPROVED** [review]

File: pkg/components/util/install.go [hunk 8]
```diff
@@ -205,7 +205,16 @@ func UninstallComponent(c components.Component, kubeconfig []byte, deleteNSBool
...
+	if err != nil {
+		return fmt.Errorf("listing Helm releases: %w", err)
+	}
+
 	if deleteNSBool {
+		if len(releasesList) > 0 {
+			return fmt.Errorf("namespace %q may have other components installed, refusing to remove", ns)
```
Comment (ipochi) [inline pkg/components/util/install.go:215]: better phrasing would be 
```
namespace `xyz` has other components installed: [list the components installed in the namespace], cannot delete namespace.
```

**Review by ipochi — DISMISSED** [review]
One suggestion, but other than that looks good.
Lovely work @knrt10 :tada: 

Glad my idea `listing helm releases in the namespace` worked out quite nicely :)

File: pkg/components/util/install.go (hunk 8 above)
Comment (invidian) [reply pkg/components/util/install.go]: We cannot really differentiate possible user's Helm releases from our releases at the moment, hence I suggested following phrasing on purpose. We cannot reliably say what components are installed there, as we do not track this information at the moment.

Comment (ipochi) [reply pkg/components/util/install.go]: True, we don't track that and that's why we also cannot delete them. I dont see it a harm to list out the releases installed in the namespace(even if they were user installed helm releases ) to show why a namespace cannot be deleted.

**Review by ipochi — APPROVED** [review]
Thank you for the PR @knrt10 :tada: 

lgtm

**Review by invidian — APPROVED** [review]

**Pull request merged**